scripts/cli/install.sh

```diff
@@ -0,0 +1,309 @@
+#!/bin/sh
+set -eu
+
+# aptos-cli's automatic install script.
+
+usage() {
+  arg0="$0"
+  cath << EOF
+Installs Aptos CLI.
+
```

Comment:
I appreciate us trying to make a script for this.

But, if we're going forward with installing the CLI automatically, we should look forward into package managers e.g. brew, apt, etc. to install these corresponding things.

The reason behind the libsso problem on newer versions of ubuntu is because we're building on 20.04 and they're using a newer version 22.04.  We can improve these things with docs rather than building another script to maintain.

Reply:
Additionally, a shell script like this isn't going to work on Window sadly.  I'd like us to stick with the three items that we have:
1. Use package managers built into OS's
2. Use the prebuilt binaries
3. Use the build from source (which I tested on multiple architectures and OS's and worked with most of them, except OpenSUSE that used zypper)

Reply:
This script tries to install the standalone build. It can also manages package manager packages. But maintaining package manager packages is going to take us more time. Since CLI can almost work without more dependencies except for the libsso hiccup. Managing package manager packages cannot be justified. 

Something we need to consider:
1. Expand the Linux support. Right now, only Debian distro is supported. We should support the Fedora/Centos/RedHat distros. Be it standalone (preferred) or package manager
2. Build CLI with a static build. No dynamic libraries, such as libsso, are required.

Reply:
Yeah I talked earlier about the static libraries.  We should figure out how to get libsso out.

We actually do support Fedora (I built it on an EC2 instance without issue), but it has to built from source.  We should build more generic builds of the CLI so that it's not so platform specific